### PR TITLE
Upgrade Solidity to 0.8.4

### PR DESCRIPTION
**Pull request by makoto**

### CODE REVIEW

File: contracts/Buffer.sol [hunk 1]
```diff
@@ -121,7 +121,7 @@ library Buffer {
         }
 
         // Copy remaining bytes
-        uint mask = 256 ** (32 - len) - 1;
+        uint mask = 256 ** ((32 - len) - 1);
```
Comment (Arachnid) [inline contracts/Buffer.sol]: This is wrong - it should be left as-is, or made `(256 ** (32 - len)) - 1`.

Comment (Arachnid) [reply contracts/Buffer.sol]: If it's generating an error as-is you can make it unchecked.

File: test/TestAlgorithms.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
+const TestBuffer = artifacts.require("TestBuffer");
```
Comment (Arachnid) [inline test/TestAlgorithms.js]: This file should probably be called something else? And we don't really need the foreach.

**Review by Arachnid — CHANGES_REQUESTED** [review]

File: contracts/Buffer.sol (hunk 1 above)
Comment (makoto) [reply contracts/Buffer.sol]: Both `256 ** ((32 - len) - 1);`(original) and `256 ** ((32 - len) - 1)`(your suggestion) will make all tests fail. What do you mean by `make it unchecked`?

File: test/TestAlgorithms.js (hunk 2 above)
Comment (makoto) [reply test/TestAlgorithms.js]: What else would you call?

File: contracts/Buffer.sol (hunk 1 above)
Comment (makoto) [reply contracts/Buffer.sol]: Wrapped with `unchecked` block

**Review by Arachnid — APPROVED** [review]

**Pull request merged**